docs/book/v3/application-integration/usage-in-a-laminas-mvc-application.md

```diff
@@ -0,0 +1,181 @@
+
+```php
+namespace Application\Controller;
+
+use Laminas\Cache\Storage\StorageInterface;
+use Laminas\Mvc\Controller\AbstractActionController;
+
+final class IndexController extends AbstractActionController
+{
+    public function __construct(
+        private readonly StorageInterface $cache
+    ) {}
+    
+    public function indexAction(): array
+    {
+        if (! $this->cache->hasItem('example')) {
+            $this->cache->addItem('example', 'value');
+        }
+
+        echo $this->cache->getItem('example') // value;
+        
+        // …
+        
+        return [];
+    }
+}
+```
+
+## Register Controller
+
+To [register the controller](https://docs.laminas.dev/laminas-mvc/quick-start/#create-a-route) for the application, extend the configuration of the module.
+Add the following lines to the module configuration file, e.g. `module/Application/config/module.config.php`:
+
+<pre class="language-php" data-line="3,8"><code>
+namespace Application;
+
+use Laminas\ServiceManager\AbstractFactory\ConfigAbstractFactory;
+
+return [
+    'controllers' => [
+        'factories' => [
+            Controller\IndexController::class => ConfigAbstractFactory::class,
+        ],
+    ],
+    // …
+];
+</code></pre>
+
+The example uses the [config factory from laminas-servicemanager](https://docs.laminas.dev/laminas-servicemanager/config-abstract-factory/) which allows any string to be used to fetch a service from the application service container, like the name of the configured cache: `default-cache`.
+
+This means that the factory [searches for an appropriate configuration](https://docs.laminas.dev/laminas-servicemanager/config-abstract-factory/#configuration) to create the controller and to resolve the constructor dependencies for the controller class.
+
+### Add Factory Configuration For Controller
+
+Extend the module configuration file to add the configuration for the controller.
+Use the name of the cache (`default-cache`), which was previously defined in the configuration of the caches, to retrieve the related cache storage instance:
+
+<pre class="language-php" data-line="11-15"><code>
+namespace Application;
+
+use Laminas\ServiceManager\AbstractFactory\ConfigAbstractFactory;
+
+return [
+    'controllers' => [
+        'factories' => [
+            Controller\IndexController::class => ConfigAbstractFactory::class,
+        ],
+    ],
+    ConfigAbstractFactory::class => [
+        Controller\IndexController::class => [
+            'default-cache',
+        ],
+    ],
+    // …
+];
+</code></pre>
+
+## Using Multipe Caches
+
+The use more than one cache backend, the factory `Laminas\Cache\Service\StorageCacheAbstractServiceFactory` allows to define multiple cache storages.
+
+Extend the cache configuration in `config/autoload/global.php` and add more cache adapters:
```

Comment:
I'd rather suggest a dedicated "cache.global.php" as having plenty stuff in the global quickly becomes messy in bigger projects

Reply:
I'll have to see if creating another file and adding a statement that it will be read automatically will keep the whole description as simple as possible.

Reply:
It usually does, the MVC skeleton has a glob-path, same for mezzio:
https://github.com/laminas/laminas-mvc-skeleton/blob/8a4ea5f0583cb518cea62e14ab6f6b8e79a6e96d/config/application.config.php#L24
https://github.com/mezzio/mezzio-skeleton/blob/8f8394a1a4440ab532bfce3fc065d0b01cc29f21/config/config.php#L40

Reply:
No, I mean we need to explain this fact additionally in this tutorial. Therefore, an already existing file is currently being used.

Reply:
> I'd rather suggest a dedicated "cache.global.php" as having plenty stuff in the global quickly becomes messy in bigger projects

Normally this note would have to be in the getting started tutorial under a heading like "Optimization" and/or "Best Practice".

Reply:
@boesing 
The tutorial has been updated to use a separate configuration file for the caches.